src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,64 @@
 
 
 def load_arguments(self, _):
-    from azure.mgmt.redis.models import RebootType, RedisKeyType, SkuName
+    from azure.mgmt.redis.models import RebootType, RedisKeyType, SkuName, TlsVersion, ReplicationRole
+    from azure.cli.core.util import shell_safe_json_parse
     from azure.cli.command_modules.redis._validators import JsonString, ScheduleEntryList
+    from azure.cli.command_modules.redis.custom import allowed_c_family_sizes, allowed_p_family_sizes
     from azure.cli.core.commands.parameters import get_enum_type  # TODO: Move this into Knack
-    from azure.cli.core.commands.parameters import get_resource_name_completion_list, name_type, tags_type
+    from azure.cli.core.commands.parameters import get_resource_name_completion_list
 
     with self.argument_context('redis') as c:
-        c.argument('name', options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
-                   completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
+                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name_without_id_part = CLIArgumentType(overrides=cache_name, id_part=None)
+        # pylint: disable=line-too-long
+        format_type = CLIArgumentType(options_list=['--file-format'], help='Format of the blob (Currently rdb is the only supported format, with other formats expected in the future)')
+
+        c.argument('name', arg_type=cache_name)
         c.argument('redis_configuration', help='JSON encoded configuration settings. Use @{file} to load from a file.',
                    type=JsonString)
         c.argument('reboot_type', arg_type=get_enum_type(RebootType))
         c.argument('key_type', arg_type=get_enum_type(RedisKeyType))
+        c.argument('files', help='SAS url for blobs that needs to be imported', nargs='+')
+        c.argument('format', arg_type=format_type)
+        c.argument('file_format', arg_type=format_type)
+        c.argument('container', help='SAS url for container where data needs to be exported to')
+        c.argument('prefix', help='Prefix to use for exported files')
+        c.argument('cache_name', arg_type=cache_name)
+        c.argument('shard_count', type=int, help='The number of shards to be created on a Premium Cluster Cache.')
+        # pylint: disable=line-too-long
+        c.argument('subnet_id', help='The full resource ID of a subnet in a virtual network to deploy the redis cache in. Example format /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1')
+        c.argument('static_ip', help='Required when deploying a redis cache inside an existing Azure Virtual Network')
+        c.argument('tenant_settings', type=JsonString, help='Json dictionary with tenant settings')
```

Comment:
JSON strings are highly discouraged. Tags syntax (--tags foo=bar key=val) is preferred. If the tenant settings are more complex than that, reach out to us to brainstorm better alternatives.